src/widget/wglwidgetqopengl.h

```diff
@@ -35,6 +36,17 @@ class WGLWidget : public QWidget {
 
     void showEvent(QShowEvent* event) override;
 
+    // By default the QOpenGLWindow will send its event to its owner
+    // WGLWidget. This function allows to override that with a different
+    // receiver. This is used for WWaveformViewer.
+    void setEventReceiver(QWidget* target) {
+        m_pEventReceiver = target;
+    }
+
+    QWidget* eventReceiver() const {
+        return m_pEventReceiver;
+    }
+
```

Comment:
whats the ownership relationship here? Is this non-owning, if so what do the lifetime requirements look like?